### ISSUE TITLE
Incorrect docs for wildcard pattern

### ISSUE THREAD

**Issue by stramel** [frontarm/navi#76]
https://frontarm.com/navi/en/guides/authenticated-routes/

```
mount({
  '/login': lazy(() => import('./login')),

  '*': map(({ context }) =>
    !context.currentUser
      ? redirect(
          '/login?redirectTo='+
          encodeURIComponent(request.mountpath+request.search),
          { exact: false }
        )
      : mount({
          '/account-details': import(() => './account-details'),
          '/billing-history': import(() => './billing-history'),
        })
  )
})
```

A few things I noticed that seem incorrect:

- I believe that `mountpath` and `search` should actually be destructured from the `NaviRequest` with `context` instead of part of a request object (unless accumulating the rest into a request object)(or maybe using `(request, context)` instead?)
- `import` is taking a function I would assume this should be `lazy(() => import('./account-details'))` instead?
- When running this example with the changes for the previous 2, I get that the URL is not found.

The modified code I ran:

```
mount({
  '/login': lazy(() => import('./login')),

  '*': map(({ context, ...request }) =>
    !context.currentUser
      ? redirect(
          '/login?redirectTo='+
          encodeURIComponent(request.mountpath+request.search),
          { exact: false }
        )
      : mount({
          '/account-details': lazy(() => import('./account-details')),
          '/billing-history': lazy(() => import('./billing-history')),
        })
  )
})
```

**Comment by jamesknelson** [comment]
Good find.

Yep, your modified code is correct. I'll need to update the Navi site – will try and get it done in the next few days.

**Comment by stramel** [comment]
@jamesknelson I also couldn't get the modified code to properly load

**Comment by jamesknelson** [comment]
Okay, I'll see if I can turn this one into a live example next week sometime.

**Comment by jamesknelson** [comment]
It turns out that the issue here was actually a bug with how `redirect()` handled the `exact` option. I've just pushed out a fix with Navi 0.12.6, and have updated the documentation with a live auth example including `{ exact: false }` and the wildcard matcher. Will publish later tonight.
